clippy_lints/src/misc.rs

```diff
@@ -600,9 +601,9 @@ fn in_attributes_expansion(expr: &Expr) -> bool {
 }
 
 /// Tests whether `def` is a variable defined outside a macro.
-fn non_macro_local(cx: &LateContext<'_, '_>, def: &def::Def) -> bool {
+fn non_macro_local(cx: &LateContext<'_, '_>, def: &def::DefKind) -> bool {
```

Comment:
This probably wants to be `res: def::Res` (also update the comment).
Oh, and, small nit: there's no need to pass any of these values by reference.